### PR TITLE
Issue #2181. Added a robots.txt

### PR DESCRIPTION
**Pull request by Somsubhra**
Added a robots.txt based on whether it is production or development version.

### CODE REVIEW

File: lib/static/views.js [hunk 1]
```diff
@@ -295,6 +295,16 @@ exports.setup = function(app) {
     renderCachableView(req, res, 'about.ejs', {title: _('About'), fullpage: false});
   });
 
+  app.get("/robots.txt", function(req, res) {
+    if (isProductionEnvironment()) {
+      var robots = fs.readFileSync(path.join(__dirname, "..", "..", "resources", "robots.txt")).toString();
```
Comment (callahad) [inline lib/static/views.js]: Can we do this asynchronously?

File: lib/static/views.js [hunk 2]
```diff
@@ -295,6 +295,22 @@ exports.setup = function(app) {
     renderCachableView(req, res, 'about.ejs', {title: _('About'), fullpage: false});
   });
 
+  app.get("/robots.txt", function(req, res) {
+    cacheAllTheThings(res);
+    res.setHeader('Content-Type', 'text/plain; charset=utf8');
+    if (isProductionEnvironment()) {
+      fs.readFile(path.join(__dirname, "..", "..", "resources", "robots.txt"), function(err, data) {
```
Comment (shane-tomlinson) [inline lib/static/views.js:302]: We should be able to cache the robots.txt file, or does cacheAllTheThings take care of this?

File: resources/robots.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+User-agent: *
+Disallow: /sign_in
+Disallow: /unsupported_dialog
+Disallow: /verify_email_address
+Disallow: /add_email_address
```
Comment (shane-tomlinson) [inline resources/robots.txt:5]: Several endpoints should be added here:
- `/cookies_disabled`
- `/reset_password`
- `/confirm`
- `/complete_transition`
- `/auth`
- `/unsupported_dialog_without_watch`
- `/relay`
- `/provision`
- `/build`
- `/production`

File: lib/static/views.js (hunk 2 above)
Comment (shane-tomlinson) [inline lib/static/views.js:302]: granted, requesting robots.txt is probably not requested very frequently.

**Comment by shane-tomlinson** [comment]
Thanks for this PR @Somsubhra! Can you also add a test in `/tests/page-requests-tests.js` around line 63 where we test whether valid requests are responded to?

File: lib/static/views.js (hunk 2 above)
Comment (Somsubhra) [inline lib/static/views.js:302]: cacheAllTheThings should probably take care of caching.

**Comment by Somsubhra** [comment]
Sure, I'll add the tests. I'll follow up with a patch adding more entries to robots.txt and a test

**Comment by shane-tomlinson** [comment]
Manually merging, thanks @Somsubhra!

**Pull request closed without merging**